onnxruntime/python/tools/transformers/fusion_rotary_attention.py

```diff
@@ -323,6 +323,7 @@ def fuse(self, normalize_node, input_name_to_nodes, output_name_to_node):
 
         # qkv_nodes_1 is for LLaMA-2 Microsoft
         # qkv_nodes_2 is for LLaMA-2 Hugging Face
+        # qkv_nodes_3 is for LLaMA-2 distribute Hugging Face model
```

Comment:
Can we say `LLaMA-2 70B model` instead like the comment for `v_nodes_4`?

Reply:
for this case, AllReduce node is introduced by shardding not the 70B model, for example if we shardding 7b or 13b model we also need to use pattern3 (qkv_nodes_3), for v_nodes_4 and k_nodes_4 it is different story, these long patterns are introduced by the repeak_kv.